core/src/main/scala/org/apache/spark/network/nio/ConnectionManager.scala

```diff
@@ -375,16 +375,22 @@ private[nio] class ConnectionManager(
                 }
               }
             } else {
-              logInfo("Key not valid ? " + key)
+              logInfo("Key not valid ? " + key + " remote address: " + 
+                  key.channel().asInstanceOf[SocketChannel].socket
```

Comment:
My first reaction was that this risks an exception just for a log message. Since this work is repeated several times, how about creating a method that will return the remote address in case of a `SocketChannel` and the default `toString()` otherwise? although given the current code, it will always be a `SocketChannel`.

Can you use string interpolation here?

Finally, the second cast isn't needed is it? it does not change the `toString()` that is called.